src/Controls/src/Core/ShellToolbar.cs

```diff
@@ -140,37 +135,20 @@ void UpdateBackbuttonBehavior()
 				return;
 
 			if (_backButtonBehavior != null)
-				_backButtonBehavior.PropertyChanged -= OnBackButtonPropertyChanged;
+				_backButtonBehavior.PropertyChanged -= OnBackButtonCommandPropertyChanged;
 
 			_backButtonBehavior = bbb;
 
 			if (_backButtonBehavior != null)
-				_backButtonBehavior.PropertyChanged += OnBackButtonPropertyChanged;
-
-			void OnBackButtonPropertyChanged(object sender, System.ComponentModel.PropertyChangedEventArgs e)
-			{
-				ApplyChanges();
-
-				if (_backButtonBehavior.Command == _backButtonCommand)
-					return;
-
-				if (_backButtonCommand != null)
-					_backButtonCommand.CanExecuteChanged -= OnBackButtonCanExecuteChanged;
-
-				_backButtonCommand = _backButtonBehavior.Command;
-
-				if (_backButtonCommand != null)
-					_backButtonCommand.CanExecuteChanged += OnBackButtonCanExecuteChanged;
-			}
+				_backButtonBehavior.PropertyChanged += OnBackButtonCommandPropertyChanged;
+		}
 
-			void OnBackButtonCanExecuteChanged(object sender, EventArgs e)
-			{
-				BackButtonEnabled =
-					_backButtonCommand.CanExecute(_backButtonBehavior.CommandParameter);
-			}
+		void OnBackButtonCommandPropertyChanged(object? sender, PropertyChangedEventArgs e)
+		{
+			ApplyChanges();
```

Comment:
`ApplyChanges` update if the back button is enabled or not but also the title, or Platform Specifics, maybe we can only update here the stuff related with the back button?

Reply:
Yea, I thought about that. My concern is that things will get missed because there are related pieces of logic and trying to be too precise with what gets re-evaluated will miss things. Especially as logic is added in the future. 

For example, this code here look at the button visibility for toolbar visibility

![image](https://user-images.githubusercontent.com/5375137/206208867-01c61e38-7c42-4b18-af44-0f9529d48522.png)

 `ApplyChanges` only fires property changes to the platform if those values have changed so for the most part, it should be a light amount of work to re-evaluate.